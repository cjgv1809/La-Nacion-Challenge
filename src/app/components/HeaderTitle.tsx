interface HeaderTitleProps {
  title: string;
}

function HeaderTitle({ title }: HeaderTitleProps) {
  return (
    <div className="com-titleWithfollow hlp-marginBottom-15">
      <h1 className="com-title-section-xl hlp-marginBottom-40">{title}</h1>
    </div>
  );
}

export default HeaderTitle;
