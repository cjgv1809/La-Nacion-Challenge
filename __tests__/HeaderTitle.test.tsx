import { render, screen } from "@testing-library/react";
import HeaderTitle from "@/app/components/HeaderTitle";

describe("HeaderTitle", () => {
  it("renders a heading", () => {
    render(<HeaderTitle title="Hello, world" />);

    const heading = screen.getByRole("heading", { level: 1 });

    expect(heading).toBeInTheDocument();
  });

  it("renders the exact title passed to it", () => {
    render(<HeaderTitle title="Hello, world" />);

    const heading = screen.getByRole("heading", { level: 1 });

    expect(heading).toHaveTextContent("Hello, world");
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<HeaderTitle title="Hello, world" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
