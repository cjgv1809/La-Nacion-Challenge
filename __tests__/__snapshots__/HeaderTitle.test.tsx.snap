// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTitle matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="com-titleWithfollow hlp-marginBottom-15"
  >
    <h1
      class="com-title-section-xl hlp-marginBottom-40"
    >
      Hello, world
    </h1>
  </div>
</DocumentFragment>
`;
